conan/tools/cmake/toolchain/blocks.py

```diff
@@ -1173,6 +1301,16 @@ def remove(self, name, *args):
             del self._blocks[arg]
 
     def select(self, name, *args):
+        """
+        keep the blocks provided as arguments, remove the others, except pre-existing "variables"
+        and "preprocessor", to not break behavior
+        """
+        self._conanfile.output.warning("CMakeToolchain.select is deprecated. Use CMakeToolchain.keep"
+                                       " instead", warn_tag="deprecated")
+        to_keep = [name] + list(args) + ["variables", "preprocessor"]
+        self._blocks = OrderedDict((k, v) for k, v in self._blocks.items() if k in to_keep)
+
+    def keep(self, name, *args):
```

Comment:
some alternatives for the name:
- filter
- pick
- include
- enable

Reply:
Something more explicit like ``tc.blocs.enabled(["block1", ...])`` looks good to me.

Future conf would read ``tools.cmake.cmaketoolchain:blocks_enabled=["block1", ]``? or ``enabled_blocks``?

Reply:
`enabled_blocks` 👍

Reply:
Done, renamed to ``blocks.enabled`` (the conf will be another PR)